### PR TITLE
Gridsome / Vuex blog post

### PR DESCRIPTION
**Pull request by dozyio**
The attached PR is for a new blog post on how to integrate Vuex with modules with Gridsome. Also includes a shopping cart example. I thought I'd write a blog post as there hasn't been one for quite a while.

Also added blog.scss as there wasn't a way to align images in blog posts.

Let me know your thoughts and let me know if you need anything changing / rewording.

### CODE REVIEW

**Comment by hjvedvik** [comment]
Thanks for the blog post, @dozyio! It looks good, but I'll take a closer look one of the next days :)

File: blog/2020-06-01-integrate-vuex-with-modules/index.md [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+excerpt: 'Vuex coupled with Gridsome provides a state management pattern you can use across your static site. This post will expand on default documentation to explain how you can use Vuex to seperate your sites state into modules.'
+---
+
+## What is Vuex? Why should I use it with Gridsome?
+
+Vuex is a state management pattern and library that can be used a centralized store for your components within an application. It provides a single source of truth to your application and reduces the complications of passing events and props through multiple components.
+
+There is already documentation on [how to add Vuex to Gridsome](/docs/client-api/#appoptions) but this tutorial will extend on this so that you can modularize your Vuex state, getters, actions and mutations.
+
+You may want to use Vuex with Gridsome if for example you were building a static frontend to an ecommerce store with Vue handling the shopping cart functionality. You may also want to handle the state of a popup banner for example - this where Vuex modules come into play as you can seperate the state into individual components.
```
Comment (jbampton) [inline blog/2020-06-01-integrate-vuex-with-modules/index.md:16]:
```suggestion
You may want to use Vuex with Gridsome if for example you were building a static frontend to an ecommerce store with Vue handling the shopping cart functionality. You may also want to handle the state of a popup banner for example - this where Vuex modules come into play as you can separate the state into individual components.
```

File: blog/2020-06-01-integrate-vuex-with-modules/index.md [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+
+Gridsome needs to be configured on how to process the Markdown products. For this you'll need to set up the plugins and templates in `gridsome.config.js`.
+
+`gridsome.config.js`
+```js
+module.exports = {
+      siteName: 'Gridsome With Vuex Modules - Shopping Cart Example',
+      plugins: [
+          {
+              // Create products from markdown files
```
Comment (jbampton) [inline blog/2020-06-01-integrate-vuex-with-modules/index.md:177]:
```suggestion
              // Create products from Markdown files
```

**Comment by tomtev** [comment]
Thanks! Could you update the data + fix conflicts?

Just a heads up: We might move this to a guide section instead of as a blog post in the future, but we can publish it as a blog post now :)

**Comment by dozyio** [comment]
Pushed the spelling fixes - thought I'd do it as a blog post as there hadn't been one for a while but yes feel free to move it to a guide if you like

**Pull request closed without merging**